test/framework/src/main/java/org/elasticsearch/test/client/RandomizingClient.java

```diff
@@ -55,13 +56,14 @@ public RandomizingClient(Client client, Random random) {
             defaultPreference = null;
         }
         this.batchedReduceSize = 2 + random.nextInt(10);
+        this.maxConcurrentRequests = 1 + random.nextInt(100);
```

Comment:
can we do something that is more likely to trigger interesting behaviours such as `1 + random.nextInt(1 << random.nextInt(8));`